src/scripts/exclusion_term_expansion/exclusion_term_expansion.py

```diff
@@ -0,0 +1,299 @@
+    include_superclass=CONFIG['include_superclass'], cache_suffix: str = None,
+    terms: List[str] = None,
+    regexp: str = None
+) -> pd.DataFrame:
+    """Query ontology using SPARQL query file
+    include_superclass: See: # https://www.w3.org/TR/sparql11-query/#propertypaths
+    cache_suffix: Caches are named using the input ontology and SPARQL files. However, it can be the case that
+      there are two different queries which use these same files, but have other differences. This param is used
+      to distinguish between such cases.
+    """
+    # Basic vars
+    query_template_filename = os.path.basename(query_template_path)
+    onto_filename = os.path.basename(onto_path)
+    results_dirname = onto_filename.replace('/', '-').replace('.', '-') + \
+        '__' + query_template_filename.replace('.', '-')
+    if cache_suffix:
+        results_dirname = results_dirname + '__' + cache_suffix
+    results_dirpath = os.path.join(CACHE_DIR, 'robot', results_dirname)
+    results_filename = 'results.csv'
+    command_save_filename = 'command.sh'
+    results_path = os.path.join(results_dirpath, results_filename)
+    command_save_path = os.path.join(results_dirpath, command_save_filename)
+    instantiated_query_path = os.path.join(results_dirpath, 'query.sparql')
+    command_str = f'robot query --input {onto_path} --query {instantiated_query_path} {results_path}'
+
+    # Instantiate template
+    with open(query_template_path, 'r') as f:
+        template_str = f.read()
+    template_obj = Template(template_str)
+    if not terms and not regexp:
+        raise RuntimeError('sparql_jinja2_robot_query: Must pass either `terms` or `regexp`.')
+    elif terms and regexp:
+        raise RuntimeError('sparql_jinja2_robot_query: Must pass only 1: `terms` or `regexp`.')
+    elif terms:
+        instantiated_str = template_obj.render(
+            property_path_qualifier='*' if include_superclass else '+',
+            values=' '.join(terms))
+    elif regexp:
+        instantiated_str = template_obj.render(regexp=regexp)
+
+    # Cache and run
+    os.makedirs(results_dirpath, exist_ok=True)
+    if not (os.path.exists(results_path) and use_cache):
+        with open(instantiated_query_path, 'w') as f:
+            # noinspection PyUnboundLocalVariable
+            f.write(instantiated_str)
+        with open(command_save_path, 'w') as f:
+            f.write(command_str)
+        subprocess.run(command_str.split())
+
+    # Read results and return
+    try:
+        df = pd.read_csv(results_path).fillna('')
+    except pd.errors.EmptyDataError:
+        # remove: so that it doesn't read this from cache, though it could be that there were really no results.
+        os.remove(results_path)
+        # could also do `except pd.errors.EmptyDataError as err`, `raise err` or give option for this as a param to func
+        df = pd.DataFrame()  # could also return None
+
+    return df
+
+
+def expand_ontological_exclusions(
```

Comment:
FYI: Since the REGEX-based queries and the explicit term queries share a lot of logic, I abstracted a lot of this logic into this function.